### ISSUE TITLE
Blackhole and External Mic not working in tandem

### ISSUE THREAD

**Issue by cookiesfortune** [ExistentialAudio/BlackHole#562]
Hi,
Im trying to record internal audio as well as external mic audio, and i can't seem to get both to record at the same time. I'm also wanting to hear the internal audio through a pair of bluetooth headphones as i record, which i have added to a Multi-Output device in Audio MIDI Setup. Here are my input + output settings:
![Screenshot 2022-04-27 at 19 47 16](https://user-images.githubusercontent.com/104497032/165491436-dc9c1393-0c9c-44ab-9b3d-97809395fa81.png)
![Screenshot 2022-04-27 at 19 47 23](https://user-images.githubusercontent.com/104497032/165491466-858eaed5-5223-4ec8-a608-6bf55e45638f.png)
Thanks in advance for your help, been struggling this for a few days and it's doing my head in

**Comment by stewartknapman** [comment]
I'm having the same issue with a similar set up.

On an M1, not sure if that makes a difference.
I've tried with multiple configurations of outputs in the multi-output settings, including the default speakers being first on the list.

The screen shots show what I think I want to be doing.
 
<img width="1059" alt="Screen Shot 2022-05-10 at 2 29 54 PM" src="https://user-images.githubusercontent.com/1373290/167530851-a4e58cf1-5b9f-40a0-ae43-5aa6741a85e1.png">
<img width="1059" alt="Screen Shot 2022-05-10 at 2 29 47 PM" src="https://user-images.githubusercontent.com/1373290/167530862-8cc859f1-b7d7-48e8-a8b8-13162abb5318.png">
<img width="1059" alt="Screen Shot 2022-05-10 at 2 31 23 PM" src="https://user-images.githubusercontent.com/1373290/167530803-326ed03c-1cee-4892-bcb5-5540f8cdbfcb.png">

**Comment by Fareed-Ahmad7** [comment]
I have same issue on m1 Mac

**Comment by jzohrab** [comment]
Sanity check: are you setting your system preferences > sound > output to "BH + Bluetooth ear" ?   This will ensure that your system output is _actually_ going to BlackHole.

**Comment by amansha28** [comment]
Hi All,

I am facing the same issue in my MacBook Air M1. Here are the setup details : 

1. In MIDI settings , created an aggregate device, "Input_Mic+Blackhole" , under this I have checked Blackhole 16ch + Oneplus Bulletz Wireless Z Bluetooth  Headphones.
2. In MIDI settings , created a Multi-Output device, "Output_Multi-Output Device" , under this I have checked Blackhole 16ch + Oneplus Bulletz Wireless Z Bluetooth Headphones.
3. In the System Preferences -> Sound -> Output  : selected "Output_Multi-Output Device".
4. In the System Preferences -> Sound -> Input  : selected Input_Mic+Blackhole.

When I record the screen using QuickTime Player I choose Microphone as : Input_Mic+Blackhole

When recording is saved and played back, I am able to hear my voice clearly but nothing from the internal audio. I have tried multiple other setting too.
Currently MacOS version : macOS Monterey 12.4.

Here are the screen shots from the MIDI and Sound.

MIDI : 

<img width="1052" alt="MIDI_1" src="https://user-images.githubusercontent.com/11635170/169865015-1aba0be8-e17b-47e2-9fc3-7b7815c9da8d.png">
<img width="1052" alt="MIDI_2" src="https://user-images.githubusercontent.com/11635170/169865037-71dd13de-c394-4c99-b64a-85ffb949ee24.png">

Sound :

**Comment by amansha28** [comment]
> Sanity check: are you setting your system preferences > sound > output to "BH + Bluetooth ear" ? This will ensure that your system output is _actually_ going to BlackHole.

@jzohrab : yep even I thought so but even when I tried as per your suggestion output is same. But thanks for suggesting though. :)

**Comment by jzohrab** [comment]
Hm. When you select blackhole on its own in the midi mixer and/or in system
output, is its volume slider up to maximum?  Wonder if it is somehow just
“swallowing” the output.  :-)

**Comment by amansha28** [comment]
> Hm. When you select blackhole on its own in the midi mixer and/or in system output, is its volume slider up to maximum? Wonder if it is somehow just “swallowing” the output. :-) El El lun, may. 23, 2022 a la(s) 9:32 a. m., amansha28 < ***@***.***> escribió:
> […](#)

@jzohrab Good point, yeah I checked its automatically slide to the maximum level when I select BlackHole in the Maximum vol.

**Comment by jzohrab** [comment]
@amansha28 - hm.  Checking the README, I see this:

> If you are using a multi-output device, due to issues with macOS the Built-in Output must be enabled and listed as the top device in the Multi-Output. https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device#4-select-output-devices

Also, check that you have a good "clock source" - https://github.com/ExistentialAudio/BlackHole#airpods-with-an-aggregatemulti-output-is-not-working

**Comment by amansha28** [comment]
> @amansha28 - hm. Checking the README, I see this:
> 
> > If you are using a multi-output device, due to issues with macOS the Built-in Output must be enabled and listed as the top device in the Multi-Output. https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device#4-select-output-devices
> 
> Also, check that you have a good "clock source" - https://github.com/ExistentialAudio/BlackHole#airpods-with-an-aggregatemulti-output-is-not-working

Thanks @jzohrab I tried these solutions too. But I realised that when I start the recording with the Blackhole and play some sound on the Mac like YouTube vid or something else that gets recorded very well.
The problem starts when I am trying to record a Zoom meeting (for which I don't want others to know), I am not able to record internal sound in such case.

**Comment by jzohrab** [comment]
There are various issues outside of BlackHole's control.  eg. https://github.com/ExistentialAudio/BlackHole#what-apps-dont-work-with-multi-outputs.  (Also, this sounds like a separate issue: this issue is about "Blackhole and External Mic not working in tandem" and your setup is working in some cases but not others.)
